pytorch_lightning/metrics/classification/helpers.py

```diff
@@ -419,21 +430,22 @@ def _input_format_classification(
         top_k=top_k,
     )
 
-    top_k = top_k if top_k else 1
-
-    if case in ["binary", "multi-label"]:
+    if case in ["binary", "multi-label"] and not top_k:
         preds = (preds >= threshold).int()
         num_classes = num_classes if not is_multiclass else 2
 
+    if case == "multi-label" and top_k:
```

Comment:
multi-label and topk? how does it work?

Reply:
Top-K highest probabilities are turned into 1s, and the rest become 0s. This is actually needed for Recall@K and Precision@K metrics, at least as understood in the Information Retrieval context, to work properly.

To make it more explicit: say you have 1000 items, and you are evaluating different queries. So each query comes with ground truth, which is a subset of these 1000 items that are relevant for your query - you can represent this as (1, 1000) binary vector. Here's where the "multi-label" part comes it - many labels in this binary ground truth vector can be 1s - as many as there are relevant documents for the query.

Then, your model will compute the relevancy scores for all 1000 items - this will be (1, 1000) float vector (and you use logit to transform it to probabilities). This part is the same as with multi-class top_k, except that the values in the prediction vector do not sum up to 1. So then, if you want to compute Precision@K by converting the top K highest probability labels in the vector of relevancy scores to 1s, and the rest to 0s.